### ISSUE TITLE
Please add more examples of queries to documentation

### ISSUE THREAD

**Issue by lappalj4** [realm/realm-java#4166]
Hi, not sure where to post this as it is kind of a feature request/ question combined. 

Anyways, I've been having difficulties building complex queries with Realm, as the documentation for query modifiers such as not() is not very extensive. Personally, I find examples to be the best way to understand how something works. This is the only part in your documentation that discusses the use of not():

> Furthermore, it is possible to negate a condition with not(). The not() operator can be used with beginGroup()/endGroup() to negate sub-conditions only.

So in a given query, do I need to use not() only once to negate the whole query? Does it matter if I have it before, after or in between my other query conditions? etc...

For example, this is a query that I've been working on. I want to find objects of type MyObject, which the user has not created, has not liked and has not disliked. Likes and dislikes are arrays of user ids in MyObject. Here's what I've tried, but it doesn't seem to be working:

```
return realm
                .where(MyObject.class)
                .not()
                .equalTo("owner.id", id)
                .or()
                .equalTo("likes.value", id)
                .or()
                .equalTo("dislikes.value", id)
                .findAll();
```

I would imagine many people would appreciate some more detailed examples on how to build more complex queries. Thanks :)

**Comment by Zhuinden** [comment]
You need to open and close the group with `beginGroup` and `endGroup`, and put the `not()` operator AFTER `endGroup()` to negate the whole subcondition

**Comment by kneth** [comment]
@lappalj4 The section on queries in our documentation is only a rough overview, and we should probably expand it. 

@Zhuinden I hope you mean BEFORE.

**Comment by kneth** [comment]
@lappalj4 Our other products (Realm Cocoa and Realm JavaScript) have string based query language. See for example https://realm.io/docs/javascript/latest/#queries. It's a subset of Apple's NSPredicate. Do you think it would be easier to use such a query language instead for the fluent interface as we have now?

**Comment by bmeike** [comment]
@lappalj4 Do you have what you need?  Can you proceed now?  I have opened https://github.com/realm/realm-java/issues/4171 to track improving our doc.  I'm going to close this ticket but you should feel absolutely free to reopen it or to open a new ticket, if there is more that we can do for you

**Comment by lappalj4** [comment]
@bmeike sorry for the absence, and thanks for the replies. I managed to solve the issue in question by using notEqualTo. Good to see that you are working on improving the relevant section in the docs , and thanks a bunch for the awesome library :)

**Comment by zachwhelchel** [comment]
@kneth has there been any movement on a string based query approach for java? We have a rather complex situation were we need to support dynamically created queries from our server on both iOS and Android. We're currently contemplating writing a function to parse an NSPredicate string into a comparable java RealmQuery object for our Android app. Would love to hear if there are any other alternatives.

**Comment by Zhuinden** [comment]
The parser exists, but it is *huge* so it was never added to Realm-Java unfortunately.

I'd love to have it as an extra config param like `realm { rawQuerySupport true}` or something

**Comment by zachwhelchel** [comment]
@Zhuinden is that parser available? I'd love to not have to build it again from scratch 😁.

**Comment by bmunkholm** [comment]
It's available in realm-core. It's used by realm-js. So you could likely easier add access to the method. PR's accepted ;-) - if nothing else then as an experimental feature. We haven't decided that we want to expose it publicly yet.

After writing above, I realized it's not really great to hijack an unrelated closed issue to discuss this in ... @zachwhelchel :-) But now I wouldn't bother deleting it :-) But in the future please create a new issue if no other open issue address you point. Or if this requires further discussion.
